pallene/coder.lua

```diff
@@ -899,16 +904,18 @@ function RecordCoder:declare_newindex()
         local typ = self:field_type(field_name)
         local value = ctx:new_tvar(typ)
         local stats = {}
-        table.insert(stats, util.render([[
-            if (PALLENE_UNLIKELY(!${CHECK_TAG})) {
-                pallene_runtime_record_type_error(L, ${KEY}, ${EXPECTED_TAG}, rawtt(${SRC_SLOT}));
-            }
-        ]], {
-            CHECK_TAG = check_tag(typ, src_slot.name, ctx),
-            KEY = key.name,
-            EXPECTED_TAG = pallene_type_tag(typ),
-            SRC_SLOT = src_slot.name,
-        }))
+        if typ._tag ~= types.T.Value then
```

Comment:
I'd rather not change this code and make `check_tag` return `1` if typ is `Value`.
This will simplify most of the modifications in this file.